src/solver.cpp

```diff
@@ -68,7 +68,8 @@ void Solver::runB(const realtype tout) const {
 
 void Solver::setup(const realtype t0, Model *model, const AmiVector &x0,
                    const AmiVector &dx0, const AmiVectorArray &sx0,
-                   const AmiVectorArray &sdx0) const {
+                   const AmiVectorArray &sdx0, bool steadystate,
```

Comment:
would prefer to have steadystate as last argument in the signature (to have consistent types grouped).

Reply:
If the re-calling of init functions works, we won't need the change in signature any more. If this approach doesn't work, I will come back to this.

Reply:
This one is fixed now. Snudials doesn't offer a public API to only set the RHS method, but this can still be changed (and is changed now) in the Sundials Memory.
Usually, I'd be uncomfortable in changing stuff which is not publicly accessible. In this case however, the init function which sets the RHS function really does nothing with it but checking for NULL and setting, and beyond that the RHS is never touched again. So just changing it in the memory is safe enough, I think...